XLS-0056d-batch/README.md

```diff
@@ -0,0 +1,503 @@
+|`TransactionType`|✔️|`string`|`UInt16`|
+|`Account`|✔️|`string`|`STAccount`|
+|`Fee`|✔️|`string`|`STAmount`|
+|`Flags`|✔️|`number`|`UInt32`|
+|`RawTransactions`|!|`array`|`STArray`|
+|`TxnIDs`|✔️|`array`|`Vector256`|
+|`BatchSigners`| |`array`|`STArray`|
+
+### 2.1. `Fee`
+
+The fee for the outer transaction is:
+
+$$(n+2)*base\textunderscore fee + \sum_{innerTxns} Txn.Fee$$
+(where `n` is the number of signatures included in the outer transaction)
+
+In other words, the fee is twice the base fee (a total of 20 drops when there is no fee escalation), plus the sum of the transaction fees of all the inner transactions (which incorporates factors like higher fees for multisign or `AMMCreate`).
+
+The fees for the individual inner transactions are paid here instead of in the inner transaction itself, to ensure that fee escalation is calculated on the total cost of the transaction instead of just the overhead.
+
+### 2.2. `Flags`
+
+The `Flags` field represents the **batch mode** of the transaction. Exactly one must be specified in a `Batch` transaction.
+
+This spec supports four modes: 
+* `ALLORNOTHING` or `tfAllOrNothing` (with a value of `0x00010000`)
+* `ONLYONE` or `tfOnlyOne` (with a value of `0x00020000`)
+* `UNTILFAILURE` or `tfUntilFailure` (with a value of `0x00040000`)
+* `INDEPENDENT` or `tfIndependent` (with a value of `0x00080000`)
+
+A transaction will be considered a failure if it receives any result that is not `tesSUCCESS`.
+
+#### 2.2.1. `ALLORNOTHING`
+All or nothing. All transactions must succeed for any of them to succeed.
+
+#### 2.2.2. `ONLYONE`
+The first transaction to succeed will be the only one to succeed; all other transactions either failed or were never tried.
+
+While this can sort of be done by submitting multiple transactions with the same sequence number, there is no guarantee that the transactions are processed in the same order they are sent.
+
+#### 2.2.3. `UNTILFAILURE`
+All transactions will be applied until the first failure, and all transactions after the first failure will not be applied.
+
+#### 2.2.4. `INDEPENDENT`
+All transactions will be applied, regardless of failure.
+
+### 2.3. `RawTransactions`
+
+`RawTransactions` contains the list of transactions that will be applied. There can be up to 8 transactions included. These transactions can come from one account or multiple accounts.
+
+Each inner transaction:
+* **Must** contain the `tfInnerBatchTxn` flag (see section 3 for details).
+* **Must not** have a fee. It must use a fee value of `"0"`.
+* **Must not** be signed (the global transaction is already signed by all relevant parties). They must instead have an empty string (`""`) in the `SigningPubKey` and `TxnSignature` fields.
+
+**This field is not included in the validated transaction, nor is it used to compute the outer transaction signature(s)**, since all transactions are included separately as a part of the ledger.
+
+### 2.4. `TxnIDs`
```

Comment:
Can you point me to an explanation of how `TxnID` is computed in the XRPL? There are some [example usages](https://github.com/XRPLF/rippled/blob/9d58f11a60881b3edb5de80cae52d8b758ae30a8/src/libxrpl/protocol/STTx.cpp#L64) in rippled, but I didn't understand the factors contributing into `TxID`.

Larger concern: If I submit a `Batch` transaction on a sidechain, is it possible for a malicious actor to copy those signatures and re-submit the `Batch` transaction on the mainnet? Does the `TxID` hash value contain references to network_id similar to CTID (Compact Transaction Identifier) ?

Reply:
Also, `NetworkID` is included in the inner transactions, which is included in the hash, so it's a moot point.

Reply:
ok

Reply:
Do you know which part of documentation describes the generation of TxID? I couldn't find which exact fields are included in the transaction hash generation process

Reply:
A hash is just a `SHA512` hash of the encoding of a transaction. So every field that is encoded is included.